### PR TITLE
ARROW-9794: [C++] Add IsVendor API for CpuInfo

### PR DESCRIPTION
**Pull request by frankdjx**
Usage: cpu_info->vendor() == CpuInfo::Vendor::Intel

Signed-off-by: Frank Du <frank.du@intel.com>

### CODE REVIEW

**Comment by emkornfield** [comment]
LGTM.  I'd like to see if @pitrou has any comments through since he is more familiar with the code.

File: cpp/src/arrow/util/cpu_info.h [hunk 1]
```diff
@@ -59,6 +59,12 @@ class ARROW_EXPORT CpuInfo {
     L3_CACHE = 2,
   };
 
+  enum Vendor {
```
Comment (pitrou) [inline cpp/src/arrow/util/cpu_info.h:62]: Better use to a `enum class`:
```c++
enum class Vendor { Unknown, Intel, AMD };
```

File: cpp/src/arrow/util/cpu_info.h [hunk 2]
```diff
@@ -101,6 +107,9 @@ class ARROW_EXPORT CpuInfo {
   /// Returns the model name of the cpu (e.g. Intel i7-2600)
   std::string model_name();
 
+  /// Returns whether or not the cpu is provided by this vendor.
+  bool IsVendor(Vendor vendor) const { return vendor == vendor_; }
```
Comment (pitrou) [inline cpp/src/arrow/util/cpu_info.h:111]: It would seem more useful to expose the vendor directly:
```c++
Vendor vendor() const { ... }
```

**Comment by pitrou** [comment]
Does this handle all of Windows, Linux, macOS?

**Comment by jianxind** [comment]
> Does this handle all of Windows, Linux, macOS?

Windows and Linux. Not know how to do it for MacOS, but seems no AMD for Mac.

File: cpp/src/arrow/util/cpu_info.h (hunk 1 above)
Comment (jianxind) [reply cpp/src/arrow/util/cpu_info.h]: done

File: cpp/src/arrow/util/cpu_info.h (hunk 2 above)
Comment (jianxind) [reply cpp/src/arrow/util/cpu_info.h]: done

**Comment by jianxind** [comment]
> > Does this handle all of Windows, Linux, macOS?
> 
> Windows and Linux. Not know how to do it for MacOS, but seems no AMD for Mac.

And seems many other APIs of cpu info is not working on MacOS also. I don't think MacOS has "/proc/cpuinfo" file.

**Review by pitrou — APPROVED** [review]
+1, thank you @jianxind

**Pull request closed without merging**